### PR TITLE
Emit a GuestState telemetry event after provision

### LINKED ISSUE #920: Telemetry Event to Report Detected CPU and Memory
There are cases where customers are paying for X CPUs and Y Memory, but they are not actually getting it. This is due to their kernel being improperly configured, or just being too old. The telemetry we have reports what Hyper-V gave the customer, which only tells half the story. I would like the agent to report the values from /proc/cpuinfo and /proc/meminfo (or something comparable), so we can detect these types of conditions.

### PR DESCRIPTION
**Pull request by boumenot**
Closes #920

### CODE REVIEW

File: azurelinuxagent/common/event.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ class WALAEventOperation:
     Partition = "Partition"
     ProcessGoalState = "ProcessGoalState"
     Provision = "Provision"
+    GuestState = "GuestState"
```
Comment (boumenot) [inline azurelinuxagent/common/event.py]: I am not crazy about the name, and I am open to suggestions.

File: tests/pa/test_provision.py [hunk 2]
```diff
@@ -115,6 +119,85 @@ def test_is_provisioned_not_deprovisioned(self,
         ph.osutil.is_current_instance_id.assert_called_once()
         deprovision_handler.run_changed_unique_id.assert_called_once()
 
+    @distros()
+    @patch('azurelinuxagent.common.osutil.default.DefaultOSUtil.get_instance_id',
+        return_value='B9F3C233-9913-9F42-8EB3-BA656DF32502')
+    def test_provision_telemetry_success(self, mock_util, distro_name, distro_version,
```
Comment (boumenot) [inline tests/pa/test_provision.py]: Let me know if there's a better way to do this.

File: azurelinuxagent/pa/provision/default.py [hunk 3]
```diff
@@ -264,12 +269,51 @@ def deploy_ssh_keypairs(self, ovfenv):
...
-                    op=WALAEventOperation.Provision)
+                    op=operation)
+
+    def get_cpu_count(self):
+        try:
+            count = len([x for x in open('/proc/cpuinfo').readlines()
+                         if x.startswith("processor")])
+            return count
+        except:
+            pass
```
Comment (hglkrijger) [inline azurelinuxagent/pa/provision/default.py]: In the exception case, should we add some logging?

File: azurelinuxagent/pa/provision/default.py [hunk 4]
```diff
@@ -88,10 +89,14 @@ def run(self):
 
             self.write_provisioned()
 
-            self.report_event("Provision succeed",
+            self.report_event("Provision succeeded",
```
Comment (hglkrijger) [inline azurelinuxagent/pa/provision/default.py]: Thanks

File: azurelinuxagent/pa/provision/default.py [hunk 5]
```diff
@@ -264,12 +269,51 @@ def deploy_ssh_keypairs(self, ovfenv):
...
+                         if x.startswith("processor")])
+            return count
+        except:
+            pass
+        return -1
+
+    def get_mem_size_mb(self):
+        try:
+            for line in open('/proc/meminfo').readlines():
+                m = re.match('^MemTotal:\s*(\d+) kB$', line)
```
Comment (hglkrijger) [inline azurelinuxagent/pa/provision/default.py]: Is it always kB?

File: azurelinuxagent/pa/provision/default.py (hunk 3 above)
Comment (boumenot) [reply azurelinuxagent/pa/provision/default.py]: How about logger.verbose?

File: azurelinuxagent/pa/provision/default.py (hunk 5 above)
Comment (boumenot) [reply azurelinuxagent/pa/provision/default.py]: The kernel docs imply it is always this way, see [filesystem/proc.txt](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/filesystems/proc.txt?h=v4.15-rc3).

**Pull request merged**